### PR TITLE
Enable support for Solr5

### PR DESCRIPTION
**Pull request by elliot-sawyer**
I've been working on a 3.6 site that needs support for Solr 5. Therefore, I've branched off the 2.4.0 tag and added support for it. The changes are almost entirely in the solrconfig.xml and types.ss file and were created from a copy of the `templates/4` folder. This work carried on from some work undertaken by @zarocknz and @firesphere helped with some implementation details.

List of changes:
- solrconfig.xml: luceneMatchVersion 6.0
- solrconfig.xml: JsonUpdateRequestHandler -> UpdateRequestHandler
- solrconfig.xml: CsvUpdateRequestHandler -> UpdateRequestHandler
- solrconfig.xml: remove node containing requestHandler name="/admin"
- types.ss: remove all instances of `enablePositionIncrements="true"`
- Solr.php: enable checks for version 5
- Solr5Service: extended classes from Solr4Service

I've kept the "default" version of the module at 4 to avoid making this a breaking change, but if a user wants to use 5, they'll need to specify that in their config (example provided in docs)

I'll raise a separate pull request to enable SS4 support when I get the time to do so.

### CODE REVIEW

**Comment by michalkleiner** [comment]
#120 related

File: code/solr/Solr5Service.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+class Solr5Service_Core extends Solr4Service_Core
+{
+
+}
+
+class Solr5Service extends Solr4Service
```
Comment (robbieaverill) [inline code/solr/Solr5Service.php]: Could you please separate this into its own file?

Comment (elliot-sawyer) [reply code/solr/Solr5Service.php]: Sure, I've split versions 3 and 4 out into their own files as well. I've also changed the target

**Comment by robbieaverill** [comment]
Sweet, cheers @elliot-sawyer

**Comment by dhensby** [comment]
looks cool; I guess it just needs testing and verifying ;) Who volunteers for that?

**Comment by elliot-sawyer** [comment]
As an FYI: the version of Solr this is running against also works on 6.6.4.

**Comment by dhensby** [comment]
> As an FYI: the version of Solr this is running against also works on 6.6.4.

might be worth adding a Solr6 class, then ;)

**Comment by elliot-sawyer** [comment]
Perhaps we can treat v6 as an undocumented feature? Rather than dump more work into SS3 support, I'd rather focus future efforts on SS4

**Comment by elliot-sawyer** [comment]
Unfortunately it's more than just a stub class. There are some other differences between Solr5 and Solr6 that are beyond the scope of this pull request. For one, spellcheck collations no longer work when running Solr 6.6.4. 

Is there anything else holding this pull request up?

**Comment by madmatt** [comment]
How can we make that easier? If I (or someone else) looks at making a tandem PR to fulltextsearch-localsolr to add v5 and v6 branches, would that be useful?

We want to upgrade this to use a modern and supported version of Solr (given all versions of Solr older than 6.4 are end-of-life), but before infrastructure can change the module needs to support it - a classic chicken-egg; but it's one that is far more easily solved by extending testing support via the localsolr module.

Thoughts?

**Comment by robbieaverill** [comment]
@madmatt I think it needs someone to take charge of it from our end. If you're happy to do that then feel free to merge this PR =)

**Comment by michalkleiner** [comment]
Parallel PR to the fulltextsearch-localsolr module would be useful. It's not essential for this piece to be merged and no party interested in this PR here expressed that need, so I wouldn't want that to be a blocker for this PR.

**Comment by abstractk** [comment]
Nice work. Any idea on when this PR will be accepted into the main branch?

**Comment by NightJar** [comment]
After testing :)
Currently no timeline on that though sorry.

**Comment by robbieaverill** [comment]
I think we just need some docs for this (see [my comment above](https://github.com/silverstripe/silverstripe-fulltextsearch/pull/226#issuecomment-454448939)).

@elliot-sawyer would you like to add a note somewhere saying Solr 5 support is experimental?

**Comment by michalkleiner** [comment]
@elliot-sawyer would you have time to finalise this so it can get merged?

**Comment by Firesphere** [comment]
15 comments. 15 comments over a 1 year period, for a support change for a better Solr version, yet, it's discussed as if it is removing a kidney from a panda?

**Comment by robbieaverill** [comment]
@Firesphere I think we're just waiting for some docs. Solr 5 support is currently not the target for support in this module which is unfortunately why we haven't dedicated any of SilverStripe Ltd's time to supporting it yet. If you'd like, we'd welcome a contribution to help get this over the line. Otherwise you're more than welcome to fork the module and customise it to your heart's content.

**Comment by Firesphere** [comment]
@robbieaverill 

No, no you're not. You're still waiting for a panda to have that operation. Docs aren't the problem and it's widely proven that Solr4 is EOL. Given it's literally out of life.

I've stopped thinking of forking the module, and at the time of writing that, you knew very well of the existence of my module, did you? If you didn't... here's a wake-up call. https://github.com/Firesphere/silverstripe-solr-search

IDGAF, to be honest, but claiming you didn't do jack shit, for a simple XML change, is a bit over the top.

**Comment by Firesphere** [comment]
Ow yeah. Hmm, by the way... I will not _ever_ endorse or support this module, because it's like a polished turd. And I don't like turds, generally speaking. So, enjoy your turd? I think?

**Comment by Firesphere** [comment]
@elliot-sawyer Just so you know. I'd say, retract this PR is probably the best way to go.

**Comment by elliot-sawyer** [comment]
I'm going to mute myself from this thread because it isn't relevant to anything I'm working on anymore. This has worked for over a year on more than one production site with zero issues to date, but we've moved on because Solr 5 is no longer receiving security patches or maintenance updates. I have neither the time nor the motivation to contribute additional docs for it because it's a bit redundant at this point. I suggest merging as-is or closing the pull request. 

If anyone else wants to use it, the work is available on my fork.

**Comment by emteknetnz** [comment]
Hi, it seems as though this pull-request is unlikely to be merged into the main code base, so we have decided to close this one.

Feel free to open a new pull-request for anything else you think could be fixed or improved.

**Pull request closed without merging**